packages/react/src/components/AccountSettings/ChangePassword/ChangePassword.tsx

```diff
@@ -1,53 +1,163 @@
   DefaultNewPassword,
   DefaultSubmitButton,
 } from './defaultComponents';
 import { ChangePasswordProps } from './types';
-import { FormValues } from '../types';
 
 const logger = new Logger('ChangePassword');
 
+const getIsDisabled = (
+  formValues: FormValues,
+  validationError: ValidationError
+): boolean => {
+  const { currentPassword, newPassword, confirmPassword } = formValues;
+
+  const hasEmptyField = !currentPassword || !newPassword || !confirmPassword;
+
+  const arePasswordsInvalid =
+    validationError.newPassword?.length > 0 ||
+    validationError.confirmPassword?.length > 0;
+
+  return hasEmptyField || arePasswordsInvalid;
+};
+
 function ChangePassword({
   onSuccess,
   onError,
+  validators,
 }: ChangePasswordProps): JSX.Element | null {
   const [errorMessage, setErrorMessage] = React.useState<string>(null);
   const [formValues, setFormValues] = React.useState<FormValues>({});
-
+  const [blurredFields, setBlurredFields] = React.useState<BlurredFields>({});
+  const [validationError, setValidationError] = React.useState<ValidationError>(
+    {}
+  );
   const { user, isLoading } = useAuth();
 
-  /** Return null if Auth.getCurrentAuthenticatedUser is still in progress  */
-  if (isLoading) {
-    return null;
-  }
+  const isDisabled = getIsDisabled(formValues, validationError);
 
-  /** Return null if user isn't authenticated in the first place */
-  if (!user) {
-    logger.warn('<ChangePassword /> requires user to be authenticated.');
-    return null;
-  }
+  const passwordValidators: FieldValidator[] = React.useMemo(() => {
+    return validators ?? getDefaultPasswordValidators();
+  }, [validators]);
+
+  /**
+   * Note that formValues and other states are passed in as props so that
+   * it does not depend on whether or not those states have been updated yet
+   */
+  const validateNewPassword = React.useCallback(
+    (
+      formValues: FormValues,
+      blurredFields: BlurredFields,
+      eventType: InputEventType
+    ): string[] => {
+      const { newPassword } = formValues;
+
+      return runFieldValidators(
+        newPassword,
+        passwordValidators,
+        eventType,
+        blurredFields.newPassword
+      );
+    },
+    [passwordValidators]
+  );
+
+  const validateConfirmPassword = React.useCallback(
+    (
+      formValues: FormValues,
+      blurredFields: BlurredFields,
+      eventType: InputEventType
+    ): string[] => {
+      const { newPassword, confirmPassword } = formValues;
+
+      const confirmPasswordValidator = getConfirmPasswordValidator(newPassword);
+
+      return runFieldValidators(
+        confirmPassword,
+        [confirmPasswordValidator],
+        eventType,
+        blurredFields.confirmPassword
+      );
+    },
+    []
+  );
+
+  const runValidation = React.useCallback(
+    (
+      formValues: FormValues,
+      blurredFields: BlurredFields,
+      eventType: InputEventType
+    ) => {
+      const newPasswordErrors = validateNewPassword(
```

Comment:
`newPasswordErrors` kind of reads as though there were previous password errors

Reply:
Changed to `passwordErrors`, think that's clear enough.